sdk/core/azure-core/CMakeLists.txt

```diff
@@ -56,10 +56,12 @@ set(
     inc/azure/core/internal/diagnostics/log.hpp
     inc/azure/core/internal/hkeyholder.hpp
     inc/azure/core/internal/http/pipeline.hpp
+    inc/azure/core/internal/http/test_transport.hpp
```

Comment:
This is internal. I will be using it from many tests. How else would I mock the HTTP response without huge code repetition. This is a convenient and simple class.

Reply:
Its fine that its internal and used for tests.  It should reside in the test directory.  See TransportAdapter for an example on to replace the replace transport adapter in tests.

Reply:
I can't have it in test directory without repetition, because it lives in Core, and I consume it from Identity. We don't have a mechanism to include test headers from other library.